docs/release.md

```diff
@@ -0,0 +1,77 @@
+* Plugin code release
+* Config release
+* VM image release
+
+Update might be one or any combination of all dimensions. But individual upgrade path is validated using adminAPI. 
+Plugin code release is a new branch and considered a major release. Any VM image or config change is a minor release.
+
+# Versioning
+
+## Major releases
+
+Major release creation flow:
+1. Create a branch with a plugin code you wish to release.
+2. Commit to update `PluginConfig` referencing release image versions.
+3. Git tag major release version (`v3.0`)
+4. Build release images (sync, metricsbridge, etc)
+5. Make sure VM image is released and is available to be used by plugin code.
+6. Update docs & release notes
+
+## Minor releases
+
+If there is a need to update already released code - minor release is done.
+This might happen if we make a mistake, want to update configuration, or not code related fix.
+
+Minor release creation flow:
+1. Add commit to release branch with required changes.
+2. Git tag minor version (`v3.1`)
+3. Build release images (sync, metricsbridge, etc)
+
+Branching model:
+```
+-------master--*----------*-----------------------
+               \          \
+                \           \-release-v4----T(v4.0)-----T(v4.1)-
+                 \-release-v3------T(v3.0)----
+```
+Note: Branch creation doesn't correspond to release, tags does.
+
+# Plugin Config / Plugin Template 
+
+When initiating plugin code and calling [GenerateConfig](https://github.com/openshift/openshift-azure/blob/master/pkg/api/plugin.go#L106). method we need to
+pass in [Plugin Template/PluginConfig](https://github.com/openshift/openshift-azure/blob/master/pkg/api/plugin/api/config.go#L8) as a parameter
+
+Plugin template defines main characteristics of the cluster. In example:
+```yaml
+imageOffer: osa
+imagePublisher: redhat
+imageSku: osa_311 
+imageVersion: 311.43.20181121 # Node VM image version
+images:
+  alertManagerBase: registry.access.redhat.com/openshift3/prometheus-alertmanager # operators base images 
+  ansibleServiceBroker: registry.access.redhat.com/openshift3/ose-ansible-service-broker:v3.11.43 # specific image version
+```
+
+Config file examples can be found in `pluginconfig/pluginconfig-{version}.yaml`. A config file is used to simplify production configuration change without new binary rollout. Struct contains secrets and it may not be wanted to store these in a yaml file.
+
+#  Architecture consideration / Topics
+
+1. Hourly image sync job should be smart enough to detect tags and build "sync list" based on those.
+   
+2. CI infrastructure should be able to rebuild all container images retrospectively in case of CI cluster rebuild.
+   Q: How we can build container images using git tags instead of branches using CI-Operator?
+   Looks like this is not possible, and we might need to develope way to do this using OpenShift build
+   Potentially customer build generator to generate build based on tags. 
```

Comment:
Yes, I think this is what we need. ci-operator can build from tags but we should consider avoiding prow altogether since these builds are one-off.